File: backend/app/schemas/reports.py
from pydantic import BaseModel


class DashboardReport(BaseModel):
    pending_deliveries_count: int
    pending_payments_count: int
    total_revenue: float


class CustomerRevenueReport(BaseModel):
    customer_id: int
    customer_name: str
    total_revenue: float


class StockReport(BaseModel):
    product_id: int
    product_name: str
    total_stock: int
    reserved_stock: int
    available_stock: int

File: frontend/src/api/reports.ts
import apiClient from './client'

export interface DashboardReport {
  pending_deliveries_count: number
  pending_payments_count: number
  total_revenue: number
}

export interface CustomerRevenueReport {
  customer_id: number
  customer_name: string
  total_revenue: number
}

export interface StockReport {
  product_id: number
  product_name: string
  total_stock: number
  reserved_stock: number
  available_stock: number
}

export const reportsApi = {
  getDashboard: async (): Promise<DashboardReport> => {
    const response = await apiClient.get('/reports/dashboard')
    return response.data
  },

  getCustomerRevenue: async (): Promise<CustomerRevenueReport[]> => {
    const response = await apiClient.get('/reports/customer-revenue')
    return response.data
  },

  getStock: async (): Promise<StockReport[]> => {
    const response = await apiClient.get('/reports/stock')
    return response.data
  },
}
